packages/rocketchat-lib/client/lib/cachedCollection.js

```diff
@@ -337,8 +337,9 @@ class CachedCollection {
 			this.log('record received', t, record);
 			RocketChat.callbacks.run(`cachedCollection-received-${ this.name }`, record, t);
 			if (t === 'removed') {
+				const type = FlowRouter.current().route.name === 'channel' ? 'c' : 'p';
```

Comment:
Using the FlowRouter inside the CachedCollection class is not safe, as the record here may not be the same room the user is viewing.